### ISSUE TITLE
[Question] Event Emitter not firing

### ISSUE THREAD

**Issue by GleytonLima** [formio/angular#560]
Hello, i make a stackblitz example to see Outputs/Events (https://github.com/formio/angular-formio/wiki/Form-Renderer#outputs-events) in action.

https://stackblitz.com/edit/angular-formio-ovgltx

But some events appears not be firing, like (invalid), (formLoad), (errorChange).

In addition,  in the first (change) event, the value isValid is true, but the form actualy is invalid, because component 'name' is required.

Maybe I'm missing something to make it work properly?

**Comment by JRiggenbach** [comment]
@GleytonLima Any progress on this?  We've noticed similar behavior?

**Comment by GleytonLima** [comment]
@JRiggenbach for now i'm using only the onChange and Submit events. For workaround, after load the form i assume It as invalid (disabling my custom Button "Save") and checking If is valid only after some change. That way i dont need to use the "Invalid" event.

**Comment by mdiazp** [comment]
> @JRiggenbach for now i'm using only the onChange and Submit events. For workaround, after load the form i assume It as invalid (disabling my custom Button "Save") and checking If is valid only after some change. That way i dont need to use the "Invalid" event.

How do you check tlif it is valid ?

**Comment by dennitorf** [comment]
Hi team! Any solution about this? I am in the same situation. I used a workaround, and I am forcing the user to make any change in the form in order to know if the form is valid or not. And, another issue is that when the event is fired the first time, the form is not valid, but is coming valid on the event.

**Comment by dennitorf** [comment]
> @JRiggenbach for now i'm using only the onChange and Submit events. For workaround, after load the form i assume It as invalid (disabling my custom Button "Save") and checking If is valid only after some change. That way i dont need to use the "Invalid" event.

I tried with this approach, but , the custom event for save is triggering twice. You faced this issue?

**Comment by SGhantous** [comment]
> @JRiggenbach for now i'm using only the onChange and Submit events. For workaround, after load the form i assume It as invalid (disabling my custom Button "Save") and checking If is valid only after some change. That way i dont need to use the "Invalid" event.

I'm also wondering how to check if the form is valid. Currently when I make a change in one of my fields that triggers a validation error (required for example), the form in no indicates that it is Invalid other than the message under the field with an error. I should mention this is with Custom Components.

**Comment by alexrok21** [comment]
Hi everyone, I'm stuck at the same point trying to emit events. I need to emit editGridSaveRow, so far this is working `<formio
		[form]='form'
		[submission]='submission'
></formio>`
but when I try this `<formio
		[form]='form'
		[submission]='submission'
		(editGridSaveRow)="test()"
	></formio>` it's not, what I'm missing? any help would be welcomed, thanks

**Comment by ZenMasterJacob20011** [comment]
Closing this thread as it is outdated. Please re-open if it is still relevant. Thank you for your contribution!
